spec/models/user_spec.rb

```diff
@@ -32,12 +32,42 @@
 RSpec.describe User, type: :model do
   subject(:user) { build(:user) }
 
+  it { is_expected.to validate_presence_of(:first_name).with_message('Enter a first name') }
+  it { is_expected.to validate_presence_of(:last_name).with_message('Enter a last name') }
+  it { is_expected.to validate_length_of(:first_name).is_at_most(40).with_message('First name must be 40 characters or less') }
+  it { is_expected.to validate_length_of(:last_name).is_at_most(40).with_message('Last name must be 40 characters or less') }
+  it { is_expected.to validate_length_of(:email).is_at_most(80).with_message('Email must be 80 characters or less') }
+  it { is_expected.to allow_value('email@addresse.foo').for(:email) }
+  it { is_expected.not_to allow_value('foo').for(:email) }
+  it { is_expected.to validate_length_of(:password).is_at_least(8).with_message('Email must be at least 8 characters') }
+
+  context 'when host is not api-sandbox' do
+    around do |example|
+      with_env('production') { example.run }
+    end
+
+    it { is_expected.not_to validate_acceptance_of(:terms_and_conditions) }
+  end
+
+  context 'when host is api-sandbox' do
+    around do |example|
+      with_env('api-sandbox') { example.run }
+    end
+
+    context 'when creating' do
+      it {
```

Comment:
nit

Rubocop would want this to be

```ruby
it do
  ...
end
```

except that this file is currently excluded for `Style/BlockDelimiter' in `.rubocop_todo.yml`

Reply:
i'll do this peter - in the middle of squashing anyway